File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        // String[] words = s.split("\\s+");
        // StringBuilder sb = new StringBuilder();

        // for (int i = words.length - 1; i >= 0; i--) 
        // {
        //     sb.append(words[i]);
        //     sb.append(" ");
        // }
        // return sb.toString().trim();


        String[] words = s.split("\\s+");
        int i=0;
        int j=words.length-1;

        while(i<j)
        {
            String temp = words[i];
            words[i] = words[j];
            words[j] = temp;

            i++;
            j--;
        }

        StringBuilder sb = new StringBuilder();
        for(String word : words)
        {
            sb.append(word).append(" ");
        }

        return sb.toString().trim();


    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> words ;
        string temp = "";

        for(auto ch : s)
        {
            if(ch != ' ')
            {
                temp+=ch;
            }
            else //ch == ' 
            {
                if(temp!="")
                {
                    words.push_back(temp);
                    temp = "";
                }
            }
        }
        if(temp!="")
        {
            words.push_back(temp);
            temp = "";
        }

        // temp = "";


       //basically ye java ka split func manually bnaya


        for(int i = words.size()-1; i>=0; i--)
        {
            temp+=words[i];
            if(i > 0)
            {
                temp+=" ";
            }
        }
        return temp;
        

        
    }
};